libs/utils/decorate.js

```diff
@@ -56,7 +56,8 @@ export function decorateBlockText(el, config = ['m', 's', 'm'], type = null) {
     if (headings) {
       headings.forEach((h) => h.classList.add(`heading-${config[0]}`));
       if (config[2]) {
-        headings[0]?.previousElementSibling?.classList.add(`detail-${config[2]}`);
+        const prevSib = headings[0]?.previousElementSibling;
+        prevSib?.classList.toggle(`detail-${config[2]}`, !prevSib.querySelector('picture'));
```

Comment:
Am I reading this right?  Toggle detail class if header.prevSib is not a picture? 
It's a little confusing and since this a shared func(), has this been tested across the rest of the blocks?

Reply:
Nope, it checks if the previous sibling _contains_ a <picture> (what will eventually be given the class of `icon-area`), and only applies `detail` styles if it does not.

This is going to be incredibly important as we move forward with "product lockups", as there are many use-cases where the lockup will be the header's previous sibling, and should absolutely not have detail styles applied to it.

In fact, this is already happening across the board -- it's just gone unnoticed because until now, icon-areas did not contain any text.

I've tested using a copy of your `kitchen-sink` doc, but am happy to do more if need be :)

Reply:
ok, that makes sense, I haven't seen many layouts having the lockup as a header sibling but It does seem a likely scenario. There may be some alignment on the `icon-area` vs 'lockup-area` definitions. 

Initially the `icon-area` was there to support `:icon-name: Label` text... for flex-gap. But some of that logic was also scoped as such for lockup and other icon size scenarios. Any-hoo the hope is moving forward for consistency, we can identify a given blocks true `lockup-area` and the global iconography.css will mostly take care of the rest. This block looks to be a little tricky w/ diff patterns and defaults. It would be good to see those examples w/ 'lockups' here if they're being included. 

I looked at a few test pages and looks good but do post up the link of the mentioned sink if you have one. Thanks

Reply:
Sure thing, I just added a bunch of stuff (non-notifications) to make sure the before and after looked the same:

[Before](https://main--milo--adobecom.hlx.page/drafts/ebartholomew/kitchen-sink)

[After](https://ebartholomew-mwpw-155529-notification-parity--milo--adobecom.hlx.page/drafts/ebartholomew/kitchen-sink)